whyphi/portal
# feat: Data Analytics Dashboard on a Listing

## What does this PR do?



This is the initial draft of a data analytics feature that is linked to each ListingCard. It enables users to visualize the distribution of key metrics for PCT applicants (e.g. `gpa`, `major`, `minor`, `class`...) using `recharts` pie charts. This branch requires an additional dependency (`recharts`), so be sure to run `npm install` in the `portal` directory in order to use this feature.


## Type of change

- [ ] Refactor: Any code refactoring
- [ ] Chore: technical debt, workflow improvements
- [x] Feature: New feature (non-breaking change which adds functionality)
- [ ] Documentation: This change requires a documentation update

## Tests Performed



So far, only some basic preliminary tests have been performed, but they appear to be successful. To test for yourself:

1. Either create a new `Listing` or use an old one (be sure that the listing is populated with applicants)
2. Click on the `settings` icon on the lisitng
3. Click `insights`
4. Go through dropdown menu to render each pie chart

## Screenshots



## Additional Comments
hehe my first PR draft 👹
